hpobench/benchmarks/mo/adult_benchmark.py

```diff
@@ -0,0 +1,439 @@
+                 training_cost : float,
+                 elapsed_time : float,
+                 DSO : float,
+                 DEO : float,
+                 DFP : float,
+                 test_DSO : float,
+                 test_DEO : float,
+                 test_DFP : float,
+                 fidelity : int,
+        """
+        self.rng = rng_helper.get_rng(rng=rng, self_rng=self.rng)
+        if shuffle:
+            self._shuffle_data(rng=self.rng, shuffle_valid=False)
+
+        ts_start = time.time()
+
+        budget = fidelity['budget']
+        logger.debug(f"budget for evaluation of config:{budget}")
+        logger.debug(f"config for evaluation:{configuration}")
+
+        sensitive_rows_val = self.X_valid[:, self.feature_names.index(self.sensitive_feature)]
+        sensitive_rows_test = self.X_test[:, self.feature_names.index(self.sensitive_feature)]
+
+        X_train, X_valid, X_test = self.X_train.copy(), self.X_valid.copy(), self.X_test.copy()
+
+        # Normalize data
+        scaler = get_fitted_scaler(X_train, "Standard")
+        if scaler is not None:
+            X_train = scaler(X_train)
+            X_valid = scaler(X_valid)
+            X_test = scaler(X_test)
+
+        # Create model
+        hidden = [configuration['fc_layer_0'], configuration['fc_layer_1'],
+                  configuration['fc_layer_2'], configuration['fc_layer_3']][:configuration['n_fc_layers']]
+
+        for item in ['fc_layer_0', 'fc_layer_1', 'fc_layer_2', 'fc_layer_3', 'n_fc_layers']:
+            configuration.pop(item)
+
+        # We deviate here from the original implementation. They have called `budget`-times mlp.partial_fit().
+        # We call `.fit()` due to efficiency aspects.
+        mlp = MLPClassifier(**configuration, hidden_layer_sizes=hidden, shuffle=shuffle,
+                            random_state=self.rng, max_iter=budget)
+        start = time.time()
+        mlp.fit(X_train, self.y_train)
+        y_pred_train = mlp.predict(X_train)
+        train_accuracy = accuracy_score(self.y_train, y_pred_train)
+        training_cost = time.time() - start
+
+        start = time.time()
+        y_pred_valid = mlp.predict(X_valid)
+        val_accuracy = accuracy_score(self.y_valid, y_pred_valid)
+        eval_valid_runtime = time.time() - start
+
+        start = time.time()
+        y_pred_test = mlp.predict(X_test)
+        test_accuracy = accuracy_score(self.y_test, y_pred_test)
+        eval_test_runtime = time.time() - start
+
+        val_statistical_disparity = fairness_risk(X_valid, self.y_valid, sensitive_rows_val, mlp, STATISTICAL_DISPARITY)
+        val_unequal_opportunity = fairness_risk(X_valid, self.y_valid, sensitive_rows_val, mlp, UNEQUAL_OPPORTUNITY)
+        val_unequalized_odds = fairness_risk(X_valid, self.y_valid, sensitive_rows_val, mlp, UNEQUALIZED_ODDS)
+
+        test_statistical_disparity = fairness_risk(X_test, self.y_test, sensitive_rows_test, mlp, STATISTICAL_DISPARITY)
```

Comment:
For the maximal network size (4 layers à 32 units), the evaluation per split takes around 0.05 to 0.1 seconds.
I would suggest to include the metrics for all splits, since it is not that expensive. 

And it might be interesting for later evaluations.